modules/library/referencing/src/test/java/org/geotools/referencing/operation/projection/GeostationarySatelliteTest.java

```diff
@@ -0,0 +1,354 @@
+        assertThat(p, is(notNullValue()));
+        geogToSphericalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMinY());
+        sphericalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToSphericalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        // Inscribed rectangle is smaller than largest inscribing rectangle, hence ESTIMATE
+//        final double tickle = 1;
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMaxX() + tickle, inscribed.getMaxY() + tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMaxX() + tickle, inscribed.getMinY() - tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMinX() - tickle, inscribed.getMaxY() + tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMinX() - tickle, inscribed.getMinY() - tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+    }
+
+    @Test
+    public void testInscribeFullDiskEstimate_Ellipsoidal() throws TransformException, FactoryException {
+
+        final Envelope2D inscribed = GeostationarySatellite.inscribeFullDiskEstimate(ellipsoidalGeosCRS);
+        assertThat(inscribed, is(notNullValue()));
+
+        final DirectPosition2D p = new DirectPosition2D();
+
+        p.setLocation(inscribed.getMaxX(), inscribed.getMaxY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMaxX(), inscribed.getMinY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMaxY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMinY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        // Inscribed rectangle is smaller than largest inscribing rectangle, hence ESTIMATE
```

Comment:
We try not to have commented out code. But since it's here I have to ask, why was it commented out?